### PR TITLE
More misc fdio

### PR DESCRIPTION
**Pull request by cgwalters**

### CODE REVIEW

File: glnx-fdio.c [hunk 1]
```diff
@@ -387,6 +387,29 @@ glnx_link_tmpfile_at (GLnxTmpfile *tmpf,
...
+ * @dfd: File descriptor for origin directory
+ * @path: Pathname, relative to @dfd
+ * @error: Error
+ *
+ * Use openat() to open a file, with flags `O_RDONLY | O_CLOEXEC | O_NOCTTY`.
+ * Like the other libglnx wrappers, will use `TEMP_FAILURE_RETRY` and
+ * also includes @path in @error in case of failure.
+ */
+gboolean
+glnx_openat_read (int             dfd,
```
Comment (jlebon) [inline glnx-fdio.c]: Can we call it e.g. `glnx_openat_rdonly` ? `openat_read` makes me think it actually reads the file as well.

File: glnx-fdio.c [hunk 2]
```diff
@@ -822,12 +845,9 @@ glnx_file_copy_at (int                   src_dfd,
       goto out;
     }
 
-  src_fd = TEMP_FAILURE_RETRY (openat (src_dfd, src_subpath, O_RDONLY | O_CLOEXEC | O_NOCTTY | O_NOFOLLOW));
-  if (src_fd == -1)
-    {
-      glnx_set_error_from_errno (error);
-      goto out;
-    }
+  /* Possibly consider O_NOFOLLOW here? Not sure it's worth it.*/
```
Comment (jlebon) [inline glnx-fdio.c]: Or make it another arg to the function? Seems like changing this behaviour has the potential to introduce some subtle regressions.

File: glnx-fdio.c (hunk 1 above)
Comment (cgwalters) [reply glnx-fdio.c]: Yep, done.

File: glnx-fdio.c (hunk 2 above)
Comment (cgwalters) [reply glnx-fdio.c]: Yeah, done.  :arrow_down:

**Comment by jlebon** [comment]
👍

**Pull request merged**